### PR TITLE
added storm wiki frontend with extra themes

### PR DESCRIPTION
**Pull request by jaigouk**
### related issue 

https://github.com/stanford-oval/storm/issues/64

### changes 

- Dracula soft dark color and other light and dark themes
- Duckduckgo search. and added searxng and SEARXNG_BASE_URL env var as a fallback option
- using ollama and added gpt-4o-mini as a fallback option
- save result files as '*.md'
- add date to to top of the result file
- added arize-phoenix to trace.

### note

because this frontend is using "fallbacks", I thought it would be better to have separate example instead of overwriting the existing demo_light.

some tests files are added and checked with https://github.com/jaigouk/storm_wiki

### CODE REVIEW

**Comment by jaigouk** [comment]
storm_wiki_6

There are 6 themes. 3 light themes and 3 dark themes. 
we can add or adjust colors.

**Comment by jaigouk** [comment]
i guess we can refine this further to configure search and llm models.

- list available search engines and add configuration options if there is api key for them in secrets. and we can also define which engine will be the fallback option. 
- same for the LLMs.

File: frontend/storm_wiki/util/search.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            try:
+                # Try DuckDuckGo first
+                results = self._search_duckduckgo(query)
+            except Exception as e:
+                logger.warning(
+                    f"DuckDuckGo search failed: {str(e)}. Falling back to SearxNG."
+                )
+                try:
+                    # Fallback to SearxNG
+                    results = self._search_searxng(query)
```
Comment (jaigouk) [inline frontend/storm_wiki/util/search.py:115]: we can change this flow once we have more search engine options. and let the user to choose which engine to use.

**Comment by shaoyijia** [comment]
Previously meant to assign the team members as the reviewer, since this looks like a pretty good enhancement. Sorry for any confusion!

**Comment by jaigouk** [comment]
is it ok to reopen and fix some issues that people have?

**Comment by shaoyijia** [comment]
Of course! Sorry for previous confusion.

**Comment by shaoyijia** [comment]
Thanks! Feel free to at me once you think it's ready for review. We are happy to help merge this.

**Pull request closed without merging**